src/layers/core/path-layer/path-layer.js

```diff
@@ -153,69 +135,72 @@ export default class PathLayer extends Layer {
         positions[i++] = point[0];
         positions[i++] = point[1];
         positions[i++] = point[2] || 0;
-        positions[i++] = point[0];
-        positions[i++] = point[1];
-        positions[i++] = point[2] || 0;
       })
     );
+    positions.copyWithin(i, 0, i);
```

Comment:
+1 for using copyWithin. But maybe add a comment about how this works. Why do we need two copies of the entire path array.